mlir/lib/Dialect/Vector/Transforms/VectorDistribute.cpp

```diff
@@ -832,10 +836,25 @@ struct WarpOpTransferRead : public OpRewritePattern<WarpExecuteOnLane0Op> {
     AffineMap map = calculateImplicitMap(sequentialType, distributedType);
     AffineMap indexMap = map.compose(read.getPermutationMap());
 
-    // Distribute the mask if present.
+    // Try to delinearize the lane ID to match the rank expected for
+    // distribution.
+    SmallVector<Value> delinearizedIds;
+    if (!delinearizeLaneId(rewriter, read.getLoc(), sequentialType.getShape(),
```

Comment:
Looked at the implementation of delinearizeLaneId (not for this commit).
It looks like it is duplicating functionality from `Dialect/Utils/IndexingUtils`.
Refactoring to reuse/extend `IndexingUtils` would also be most welcome.